src/apps/mesoscale-explorer/ui/panels.tsx

```diff
@@ -6,14 +6,40 @@
 
 import { Mp4EncoderUI } from '../../../extensions/mp4-export/ui';
 import { PluginUIComponent } from '../../../mol-plugin-ui/base';
+import { AnimationViewportControls, LociLabels, SelectionViewportControls, StateSnapshotViewportControls, TrajectoryViewportControls } from '../../../mol-plugin-ui/controls';
 import { SectionHeader } from '../../../mol-plugin-ui/controls/common';
+import { BackgroundTaskProgress } from '../../../mol-plugin-ui/task';
+import { Toasts } from '../../../mol-plugin-ui/toast';
+import { Viewport, ViewportControls } from '../../../mol-plugin-ui/viewport';
 import { MesoscaleExplorerState } from '../app';
 import { MesoscaleState } from '../data/state';
-import { EntityControls, ModelInfo, SelectionInfo } from './entities';
+import { EntityControls, MesoViewportSnapshotDescription, ModelInfo, SelectionInfo } from './entities';
 import { LoaderControls, ExampleControls, SessionControls, SnapshotControls, DatabaseControls } from './states';
 
 const Spacer = () => <div style={{ height: '2em' }} />;
 
+export class MesoScaleViewport extends PluginUIComponent {
+    render() {
+        const VPControls = this.plugin.spec.components?.viewport?.controls || ViewportControls;
+        return <>
+            <Viewport />
+            <div className='msp-viewport-top-left-controls'>
+                <AnimationViewportControls />
+                <TrajectoryViewportControls />
+                <StateSnapshotViewportControls />
+                <MesoViewportSnapshotDescription />
```

Comment:
I wonder if we should should add just the ability to replace the snapshot description component to the `PluginSpec`?

Reply:
Can you do it?

Reply:
https://github.com/molstar/molstar/pull/1169